File: clients/python_pypy/model/min_max_range_long.py
from stream_wrapper import StreamWrapper

class MinMaxRangeLong:
    """Range of values"""

    __slots__ = ("min","max",)

    min: int
    max: int

    def __init__(self, min: int, max: int):
        self.min = min
        """Minimal value"""
        self.max = max
        """Maximal  value"""

    @staticmethod
    def read_from(stream: StreamWrapper) -> "MinMaxRangeLong":
        """Read MinMaxRangeLong from input stream
        """
        min = stream.read_long()
        max = stream.read_long()
        return MinMaxRangeLong(min, max)
    
    def write_to(self, stream: StreamWrapper):
        """Write MinMaxRangeLong to output stream
        """
        stream.write_long(self.min)
        stream.write_long(self.max)
    
    def __repr__(self):
        return "MinMaxRangeLong(" + \
            repr(self.min) + \
            ", " + \
            repr(self.max) + \
            ")"

File: clients/kotlin_graal/src/main/kotlin/komus24/model/MinMaxRangeLong.kt
package komus24.model

import komus24.util.StreamUtil

/**
 * Range of values
 */
class MinMaxRangeLong {
    /**
     * Minimal value
     */
    var min: Long
    /**
     * Maximal  value
     */
    var max: Long

    constructor(min: Long, max: Long) {
        this.min = min
        this.max = max
    }

    /**
     * Write MinMaxRangeLong to output stream
     */
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeLong(stream, min)
        StreamUtil.writeLong(stream, max)
    }

    /**
     * Get string representation of MinMaxRangeLong
     */
    override fun toString(): String {
        var stringBuilder = StringBuilder("MinMaxRangeLong { ")
        stringBuilder.append("min: ")
        stringBuilder.append(min)
        stringBuilder.append(", ")
        stringBuilder.append("max: ")
        stringBuilder.append(max)
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        /**
         * Read MinMaxRangeLong from input stream
         */
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): MinMaxRangeLong {
            var min: Long
            min = StreamUtil.readLong(stream)
            var max: Long
            max = StreamUtil.readLong(stream)
            return MinMaxRangeLong(min, max)
        }
    }
}